pandera/backends/pandas/array.py

```diff
@@ -55,6 +56,51 @@ def validate(
             except SchemaError as exc:
                 error_handler.collect_error(exc.reason_code, exc)
 
+        # run the core checks
+        error_handler = self.run_checks_and_handle_errors(
+            error_handler,
+            schema,
+            check_obj,
+            head,
+            tail,
+            sample,
+            random_state,
+        )
+
+        if lazy and error_handler.collected_errors:
+            if drop_invalid:
+                check_obj = self.drop_invalid_data(check_obj, error_handler)
+                return check_obj
+            else:
+                raise SchemaErrors(
+                    schema=schema,
+                    schema_errors=error_handler.collected_errors,
+                    data=check_obj,
+                )
+
+        return check_obj
+
+    def drop_invalid_data(self, check_obj, error_handler):
+        """Remove invalid elements in a `check_obj` according to failures in caught by the `error_handler`"""
+        errors = error_handler.collected_errors
+        for err in errors:
+            check_obj = check_obj.loc[
+                ~check_obj.index.isin(err.failure_cases["index"])
+            ]
+        return check_obj
+
+    def run_checks_and_handle_errors(
```

Comment:
@cosmicBboy this method _could_ be moved into the parent class to remove the duplication, but I'm not sure this would be the right move. They are quite different implementations, and don't want to abstract it to the parent for some vain DRYness 😅

Reply:
similar method in `container.py`